crates/turborepo-lib/src/commands/ls.rs

```diff
@@ -30,10 +31,42 @@ struct RepositoryDetails<'a> {
     packages: Vec<(&'a PackageName, &'a AnchoredSystemPath)>,
 }
 
+impl<'a> Serialize for RepositoryDetails<'a> {
```

Comment:
Any reason we're implementing this manually versus doing the derive?

Reply:
Using the derive gives us arrays:

```
packages: [
 [name, path]
 ...
]
```

But I wanted a little more control over it

Reply:
I would highly suggest leveraging `#[serde(into)]` [docs](https://serde.rs/container-attrs.html#into) here instead of manually implementing this. The general pattern is create a new type like `RepoDetailsDisplay` and then create a `impl<'a> From<&'a RepositoryDetails<'a>> for RepoDetailsDisplay<'a>` which does any of the desired conversions. e.g. `(&PackageName, &AnchoredSystemPath) -> PackageDetails { name: String, path: String }`

Reply:
Nice, thanks for the suggestion - done!